buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchNode.java

```diff
@@ -548,15 +548,15 @@ public synchronized void start() {
 
         logToProcessStdout("Creating opensearch keystore with password set to [" + keystorePassword + "]");
         if (keystorePassword.length() > 0) {
-            runOpenSearchBinScriptWithInput(keystorePassword + "\n" + keystorePassword, "opensearch-keystore", "create", "-p");
+            runOpenSearchBinScriptWithInput(keystorePassword + "\n" + keystorePassword + "\n", "opensearch-keystore", "create", "-p");
```

Comment:
historically opensearch-keystore tool didn't know how to correctly load BC-FJA module (and associated jars) nor able to create a keystore in `bcfks` format. Is this tool now able to require long enough password (14 characters) and create keystore in bcfks format? Note one may need to set `JAVA_TOOL_OPTIONS" : "--module-path=/path/to/dir/with/bouncycastle-fips/jars/"` or similarly use maven extract plugin to uber-jar all bcfips jars; and then extract them to a temp location.

Reply:
I agree - this circumstance is a big issue with the current branch. `KeyStoreWrapper` states that internal keystore has built-in FIPS compliance starting from ES 6.3. The keystore uses the AES/GCM/NoPadding algorithm with the PBKDF2WithHmacSHA512 key derivation function, and the maximum key length is limited to 128 bits in UTF-8.

However, when the keystore is created using opensearch-cli, it does not run in FIPS mode with the BC libraries on the classpath. As a result `KeyStoreWrapper` falls back to the built-in Sun/Oracle security providers. This situation causes issues in the current integration tests because the `$OPENSEARCH_CRYPTO_STANDARD` environment variable does not propagate in this scenario.

To address this, as defined in `opensearch-cli`, the `$OPENSEARCH_ADDITIONAL_CLASSPATH_DIRECTORIES` variable can be used to include BC libraries.

Reply:
Note otherprojects have hit this too; it would be really nice if opensearch-cli could query the default keystore type; and really try hard to use it; and fail if it is incapable of doing it.

At chainguard we do set default keystore type to bcfks, but lots of tools fail to honor it; or have ability to specify additional `--module-path` or `--classpath` to their java invocations, and/or otherwise vendor-in (uber-jar'ed-in) the bcfips jars to ensure the bcfks keystore creation "just works" upon launch as "normal".

I do question the many places that currently uses the deprecated, even in Java8, keystore type of JKS. As imho all references to JKS should be eliminated towards requesting default keystore type which in most contexts today would be pkcs12, nss/pkcs11/nssdb-fips for the likes of rhel-ubi, and bcks / bcfks for the bcfips based jdks (like the ones from chainguard).

I guess I also should finally make a public docker image of a universal keytool that "just works" and has support for all keystore types, properly, out of the box. For efficient keystore creation / import / export / migration. To avoid need for individual projects of implementing that, yet again, themselves.

Reply:
I see where you're coming from and thank you for explaining the background.

The internal KeyStore is not intended to store certificate chains; instead, it acts like a key/value map with specific requirements that should operate w/o a FIPS context. I believe this is why it was designed in that manner. At least I don't see the request for possible migration to BCFKS standard as part of this PR.

Regarding Key- & TrustStores that are used for TLS communication, their type is usually predefined. If not, it is inferred from the file extension (see `KeyStoreType`). `KeyStoreFactory` prevents non-secure KeyStore types from being instantiated. For example, if the application is running in a FIPS context but the TrustStore type has a JKS extension, a runtime exception will be thrown.